proposed/2023/InsecureConnectionsDisableCertificateValidation.md

```diff
@@ -0,0 +1,124 @@
+# Allow insecure connections and disable certificate validation
+
+- [Jon Douglas](https://github.com/jondouglas)
+- Start Date (2023-04-13)
+- [#4387](https://github.com/NuGet/Home/issues/4387), [#12015](https://github.com/NuGet/Home/issues/12015)
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+NuGet does not provide much functionality for controlling the decisions of trust against http and https sources. In recent surveys, we have even seen a general sentiment towards using 2 or more sources for their projects. Many of these sources have different security requirements that should be up to the developer to decide on. 
+
+There are two functionalities that need to be added to NuGet tooling to help companies move over to a "HTTPS Everywhere" ideal world.
+
+1. I have a HTTP server and don't want the HTTPS warning/error.
+2. I have a HTTPS server and I want to ignore SSL/TLS certification validation.
+
+In both of these cases, developers would like more fine-grain control to opt-out of these security best practices such in cases where they have their own local http source or are self-signing with a certificate.
+
+## Motivation 
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+When we blogged about an initial plan about ["HTTPS Everywhere" for NuGet](https://devblogs.microsoft.com/nuget/https-everywhere/), we were met with widespread feedback on scenarios we aren't always aware of. The primary scenario is that a company may have a HTTP source on a local network that they would like to remain using based on their other company security practices. There's also still a challenge of obtaining a SSL/TLS certificate which may take longer than the timeline we initially proposed (~3 years).
+
+With this feedback in mind, we would like to ensure that we have a generally agreed upon plan that provides these mechanism for flexibility based on where you are personally on your journey to "HTTPS Everywhere" in the [.NET 8 timeframe given it is a LTS](https://dotnet.microsoft.com/en-us/platform/support/policy). While these mechanisms are not recommended to use long-term and generally are seen as not an ideal security practice, we believe that it is not our place to push specific security ideology and rather make our tools flexible to meet everyone where they are.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+
+There will be multiple ways to opt-out of these two functionalities described above.
+
+#### Global NuGet.config
+
+A developer may choose to want a global setting applied to all sources in which they may allow insecure connections on any defined HTTP source and disable certificate validation on any defined HTTPS source.
```

Comment:
IMHO we should have the setting per source (as suggested below), and not have a global config. The reasoning is that this config is risky, and the user should consider if it's appropriate for each feed specifically. This will also prevent a situation when a new external feed is added and those settings are applied unintentionally.

Reply:
Let's leave this conversation open for more community input. Especially to hear more cases where this might be useful.

Reply:
I agree that it should be per source. I want to keep validating for example nuget.org, but not my internal server.

Reply:
I have redacted the other experiences based on feedback.